### DISCUSSION TITLE
Information about requested permissions

### DISCUSSION THREAD

**Discussion by tomastan** [libre-tube/LibreTube#1119] [General]
### Describe your suggested feature

The app requests some permissions which are not obvious enough:

<img src="https://user-images.githubusercontent.com/68420660/184479626-746fe3d2-add9-4e41-830b-326268a87050.jpg" height="600">

For the sake of transparency, could it be disclosed in the documentation why such permissions are needed (if they really are). Most questions are about:
- modify delete the contents on your **shared storage** 
- read the contents on your **shared storage** 
- run at startup
- view network connections

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
The permissions are needed for the following tasks:

1. `modify delete the contents on your shared storage`: needed for downloading videos
2. `read the contents on your shared storage`: needed for importing subscriptions
3. `run at startup`: needed for notifications for new streams
4. `view network connections`: needed to make sure you have internet connection on app startup and when searching for new streams in the background

Should I add a section to the README about it?

**Comment by tomastan** [comment]
@Bnyro Thanks for clarification.
It's your project so it is you to decide whether to add or not. :)

IMHO, it would make sense, having in mind the app is installed not from the Google Play, but as an 3rd party APK. This automatically make some people cautious, so being transparent with the security and disclose everything in advance would be a good step for building good reputation..

**Reply by martinespinoso32-hue** [reply]
Alguien puede botar por mi botar por ustedes y me pueden decir como hablo con los creadores de libretube para dar una información

**Comment by Bnyro** [comment]
Agree, transparency is important for FOSS projects.
I just have been browsing some other Android apps on Github and couldn't find any that has something like that in the README.
If we had some sort of documentation it would probably fit well there.
I'm still not sure about adding it to the README, it would probably very technical and nothing most users care about. Maybe we should just add a short privacy section which says that permissions needed don't collect any personal data or do weird stuff, idk

**Comment by tomastan** [comment]
Sorry, for some delay - digital detox :).

You see, permissions and privacy are a bit different topics here. I'm also not sure it's correct to compare with other FOSS apps, if they are available on Google Play and get trust credit that way.

When people install LibreTube as an APK - there are numerous warnings and alerts, about data could be leaked, phone broken and so on. It scares a regular user. Then it asks for permissions, and it scares again - with a permissions not very intuitive for just watching Youtube. So cautious users may be turned down by this and decide to avoid the app, that could look to be malicious. Some users who are not diverted may be interested in information why such permissions are requested. This is why I asked you as I didn't find anything about it.

So regardless a standard Privacy block in README, which not many people cares. I'd consider 3 options:
- Add short section what you answered me describing permissions, or move to a separate page.
- Leave this issue tagged and updated if something changes. 
- Request for permissions on demand. It would be self explanatory and no need to document then. It looks to be the best way from the user point of view.

These are just proposals. I understand there are more other critical issues with the app which may have higher priority for you.

**Comment by Bnyro** [comment]
Sorry for the delay, that's why we're open source ;-)
I think it's the best if I just convert this issue into a discussion and pin it in the discussions tab then which will
1. allow further discussions about
2. inform other interested users about the reasons for the permission usage

Apart from that, everybody is invited to study the source code on his own to look up what permissions are used for, it's self-explaining when reading the concerning code imo.

Greetings

**Reply by tomastan** [reply]
I agree. It could be taken as this. At least those who are interested could find this information. Thanks!

**Comment by Bnyro** [comment]
Short update: Storage permissions are not needed nor requested anymore :)

**Comment by Slomeam** [comment]
The video stops playing if you turn off the phone
